src/silhouette.jl

```diff
@@ -1,5 +1,71 @@
 # Silhouette
 
+mutable struct SilhouettesDistsPrecompute{T}
+    k::Int
+    d::Int
+    counts::Matrix{Int} #[k, 1]
+    Ψ::Matrix{T} #[k, 1]
+    Y::Matrix{T} #[d, k]
+end
```

Comment:
1. :heavy_check_mark: 
2. :heavy_check_mark: 
3. What do you think of `cluster_first_moment` and `cluster_second_moment`? I think they are more descriptive here.
4. See the example in the docstring of `silhouettes_precompute_batch!`. As you can see, the user needs to hold an instance of `SqEuclideanPrecomputedSilhouettes` as he runs through the data set. The data points fed to `silhouettes` are in general a tiny fraction of the dataset, so he needs to keep a reference to the precomputed values as he goes.
5. The purpose of the matrix vs vector was for the broadcasting operations, so that there is no ambiguity there. What of the case of i.e. `k == d`...

Reply:
3. I'm not sure I understand the code well enough to suggest the proper names. :) Too long names names also degrade code readability. Single letter vars are OK if there is a comment with the description, but for fields like k and d the names nclusters and dims would be better IMO.
4. Sure, you can have an exported method that returns this object, but there is no need to export the type of object, as the user does not need to access its fields. If you want, you may want to declare an abstract *AbstractPrecomputedSilhouettes{T}* type export it, and inherit all concrete types from it. But at the moment if it's an overkill as we have a single metric, and I suspect uses of "streaming" API wiggle be rare.
5. Sure, for broadcasting you might be required to reshape it, before using, that's what I've meant. But reshaping just creates a temporary array that references the same memory, so it's cheap. For any other operations keeping the dimensions as lean as possible is the most practical solution.

Reply:
These points should be resolved, but please take a look and resolve this conversation if all is well.